File: interview-preparation-kit/archive/arrays/left_rotation.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the rotLeft function below.
def rotLeft(a, d):
    arr = a[d:]
    arr.extend(a[:d])
    return arr

if __name__ == '__main__':

    nd = input().split()
    n = int(nd[0])
    d = int(nd[1])

    arr = list(map(int, input().rstrip().split()))

    result = rotLeft(arr, d)

    print(' '.join(map(str, result)))

File: interview-preparation-kit/archive/arrays/left_rotation.js
'use strict';

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({input: process.stdin, output: process.stdout});

let inputString = [];
let currentLine = 0;

rl.prompt();

rl.on('line', (inputLine) => {
    inputString.push(inputLine);
})

rl.on('close', () => {
    console.log(main(inputString));
    process.exit(0);
})

function rotLeft(a, d) {
    return [...a.slice(d), ...a.slice(0, d)]
}

function main(inputString) {
    const nd = inputString[0].split(' ');

    const n = parseInt(nd[0], 10);
    const d = parseInt(nd[1], 10);
    const a = inputString[1].split(' ').map(aTemp => parseInt(aTemp, 10));

    const result = rotLeft(a, d);

    return result.join(' ')+'\n'
}
